src/Psl/Str/Grapheme/reverse.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Str\Grapheme;
+
+use Psl\Exception\InvalidArgumentException;
+
+use function grapheme_strlen;
+use function grapheme_substr;
+
+/**
+ * Reverses the string.
+ *
+ * @pure
+ *
+ * @throws \Psl\Exception\InvalidArgumentException If $string length cannot be determined.
+ */
+function reverse(string $string): string
+{
+    $reversed = '';
+
+    $length = grapheme_strlen($string);
+
+    if ($length === false || $length === null) {
+        throw new InvalidArgumentException('Cannot get string length. Possibly invalid UTF-8 sequence');
+    }
+
+    while ($length-- > 0) {
+        $reversed .= grapheme_substr($string, $length, 1);
+    }
```

Comment:
handling these errors shouldn't be done here, there's already wrappers for these grapheme_* functions.

```suggestion
    $length = length($string);

    while ($length-- > 0) {
        $reversed .= slice($string, $length, 1);
    }
```